README.md
# TMF ( TCExam Mobile Friendly ) CBT For AKM Support Android

Aplikasi Ujian (CBT) ini cocok untuk berbagai mata pelajaran maupun segala jenjang pendidikan atau kebutuhan. Mampu menampung banyak peserta ujian dalam satu waktu bersamaan meskipun dijalankan pada shared hosting. 
Tipe soal cukup beragam, bisa dikondisikan untuk melatih peserta didik mempersiapkan AKM atau untuk event lainnya.
Projek ini merupakan Forking dari TMF (TCExam Mobile Friendly) dengan modifikasi pada loading soal dan kirim jawaban melalui AJAX dari sisi peserta ujian.

Halaman Login

<img src="#" width="200px"/>

Logo sekolah dan nama sekolah serta atribut lainnya dapat diubah sesuai selera.


Daftar Ujian

<img src="#" width="200px"/>

## Mendukung beberapa tipe soal seperti
- Pilihan ganda jawaban tunggal
- Pilihan ganda jawaban jamak
- Menjodohkan / mengurutkan jawaban
- Isian singkat koreksi Otomatis
- Uraian panjang dengan kemampuan tambahan upload jawaban dari gambar / foto

## Fitur improvement dari TCExam
- Input Soal lebih mudah, melalui Template MS Excel
- Disediakan layanan online utk input soal melalui Template MS Word.
- Fitur Mode Darurat (Lembar Halaman Offline), yang memungkinkan proses ujian tanpa kuota utama dan tanpa server.
- Editing soal menggunakan antar muka HTML Editor, support konversi otomatis equation dari MS Word
- Generate Test Data agar loading soal lebih cepat dengan user lebih banyak dalam satu waktu
- Mengijinkan editing soal/jawaban/setting ujian meskipun Test sedang berjalan
- Laporan rekap kehadiran Test
- Peserta ujian dapat meng-upload jawaban dalam bentuk gambar pada jenis soal Uraian
- Export hasil Test ke format EXCEL
- Halaman General Setting untuk mengatur beberapa bagian CBT dengan lebih mudah
- Opsi penjodohan bisa berupa angka maupun Teks, labelnya bisa berupa angka, teks, gambar atau media lain seperti video
- Jumlah pengulangan ujian dapat diset sesuai keinginan, tidak lagi unlimited, namun opsi unlimited masih tersedia.
- Theme yang lebih modern dan fresh :)
- Export dan Import user melalui format file Excel
- Terdapat fitur pengumuman Surat Keterangan Lulus Online, namun dengan teknik yang masih sederhana.
- Lebih gesit dan ringan, sudah dipakai pada shared hosting mampu 300an user serentak.

## Panduan
Panduan instalasi dan tutorial lainnya silakan bisa dipelajari di youtube dengan keyword searching Malas Tenar Community.

## Kebutuhan Sistem
1. Web server (Apache/nginx/OLS, dan lainnya)
2. PHP 5.6 s.d 8.0.x
3. Database MySQL/MariaDB/Postgre/Oracle

## Contact
Apabila berminat ingin bergabung dalam tim pengembangan, silakan join Author melalui saluran di bawah ini
1. Blog: [evankumanireng2.blogspot.com](https://evankumanireng2.blogspot.com)
2. Email: evankumanireng@gmail.com


# TCExam - README

------------------------------------------------------

FOR ANY USAGE THAT REQUIRES COMMERCIAL LICENSING TERMS,
PLEASE CONTACT INFO@VARIANZA.COM TO PURCHASE A LICENSE.

------------------------------------------------------

* Name: TCExam
* Author: Nicola Asuni
* URLs:
	* http://www.tcexam.org
	* https://github.com/tecnickcom/tcexam
* Requirements and Install Documentation:
	* [ENG] install/index.htm
	* Check also the documentation on http://www.tcexam.org
* Source Documentation:
	http://www.tcexam.org
* Support - Help:
	Please use support Issues at https://github.com/tecnickcom/tcexam
* Software License:
	* Copyright (C) 2004-2020 Nicola Asuni - Tecnick.com LTD
	* See LICENSE.TXT file for more information.


[![Donate via PayPal](https://img.shields.io/badge/donate-paypal-87ceeb.svg)](https://www.paypal.com/cgi-bin/webscr?cmd=_donations&currency_code=GBP&business=paypal@tecnick.com&item_name=donation%20for%20tcexam%20project)
*Please consider supporting this project by making a donation via [PayPal](https://www.paypal.com/cgi-bin/webscr?cmd=_donations&currency_code=GBP&business=paypal@tecnick.com&item_name=donation%20for%20tcexam%20project)*


## Third-party Software

TCExam includes some third-party software components that are not strictly required but have been included as you convenience, and if used are subject to their respective licenses.

* **PHPMailer**
    * Full Featured Email Transfer Class for PHP.
    * Author: Brent R. Matzelle (bmatzelle@yahoo.com)
    * Homepage: https://github.com/PHPMailer/PHPMailer
    * License: LGPL (GNU LESSER GENERAL PUBLIC LICENSE)
    * Location: /shared/phpmailer/

* **The DHTML Calendar**
    * Calendar widget written in Javascript.
    * Author: Mihai Bazon (mihai_bazon@yahoo.com)
    * Homepage: http://dynarch.com/mishoo/
    * License: LGPL (GNU LESSER GENERAL PUBLIC LICENSE)
    * Location: /shared/jscripts/jscalendar/

* **Javascript VirtualKeyboard**
    * Author: Ilya Lebedev (ilya@lebedev.net)
    * Homepage: http://debugger.ru/projects/virtualkeyboard
    * License: LGPL (GNU LESSER GENERAL PUBLIC LICENSE)
    * Location: /shared/jscripts/vk/

* **Radius Class**
    * Radius client implementation in pure PHP.
    * Author: SysCo/al (developer@sysco.ch)
    * Homepage: http://developer.sysco.ch/php/
    * License: LGPL (GNU LESSER GENERAL PUBLIC LICENSE)
    * Location: /shared/radius/

* **phpCAS**
    * Central Authentication Service (CAS) client in PHP.
    * Copyright 2007-2011, JA-SIG, Inc. http://www.jasig.org/
    * Homepage: https://wiki.jasig.org/display/CASC/phpCAS
    * License: Apache License, Version 2.0 http://www.apache.org/licenses/LICENSE-2.0
    * Location: /shared/cas/

* **Fonts**

    TCExam includes fonts for the TCPDF library, they are not strictly required but have been included as you convenience.

    All the PHP files on the fonts directory are subject to the general TCPDF license (GNU-LGPLv3),
    they do not contain any binary data but just a description of the general properties of a particular font.
    These files can be also generated on the fly using the font utilities and TCPDF methods.

    All the original binary TTF font files have been renamed for compatibility with TCPDF and compressed using the gzcompress PHP function that uses the ZLIB data format (.z files).

    The binary files (.z) that begins with the prefix "free" have been extracted from the GNU FreeFont collection (GNU-GPLv3).
    The binary files (.z) that begins with the prefix "pdfa" have been derived from the GNU FreeFont, so they are subject to the same license.
    For the details of Copyright, License and other information, please check the files inside the directory fonts/freefont-20100919
    Link : http://www.gnu.org/software/freefont/

    The binary files (.z) that begins with the prefix "dejavu" have been extracted from the DejaVu fonts 2.33 (Bitstream) collection.
    For the details of Copyright, License and other information, please check the files inside the directory fonts/dejavu-fonts-ttf-2.33
    Link : http://dejavu-fonts.org

    The binary files (.z) that begins with the prefix "ae" have been extracted from the Arabeyes.org collection (GNU-GPLv2).
    Link : http://projects.arabeyes.org/


## Description

TCExam is a software solution (CBA - Computer-Based Assessment) to
drastically simplify the evaluation process for schools, universities,
recruiting firms as well as private and public companies, allowing
professors, teachers and examiners to create, distribute and manage exams,
tests, surveys and quizzes.

Compared to the traditional Pen-and-Paper Testing (PPT), TCExam simplifies
the whole process of evaluation reducing costs and improving quality and
reliability of the examinations.

TCExam comes in a variety of languages and is already used all over the
world by top universities, schools, private and public institutions,
independent educators and research centers.

### KEY FEATURES

* **Flexibility and Configurability**: The system has been designed to offer a
high degree of adaptability to cover a great variety of usage scenarios.
Numerous configuration features allow customizing TCExam to fit all possible
requirements.

* **Free Open Source Software (FOSS)**: TCExam software is distributed with
GNU-AGPLv3 license. Open Source promotes Software reliability and quality by
supporting independent peer review and rapid evolution of the source code.

* **Web-based Architecture**: TCExam is Web-based and developed with LAMP
(Linux, Apache, MySQL, PHP) technology. User-friendly interface allows you
to install and use TCExam by simply connecting one cable to the switch of
your computer network. All the computers on the network will be able to use
TCExam independently from the client’s operating system as a normal intranet
website, with no need to install any additional software or plug-in.

* **Internationalization (I18N)**: TCExam is language-independent through the
adoption of the UTF-8, Unicode and  TMX standards. It supports the
Right-To-Left mode and currently includes translations in 25 different
languages.

* **Accessibility and Usability**: TCExam Web interface conforms to the XHTML
1.0 Strict standard, the guidelines on Accessibility (W3C-WAI-WCAG 1.0) and
Usability, to provide equal access and equal opportunity to people with
disabilities, including blindness.

* **Results and Statistics**: TCExam outputs a variety of result pages, enabling
various selection filters and providing numerous statistical indexes.
Results and statistics can then be exported into various formats for filing
or reworking. The test-takers can immediately be informed of the result of
their test, or have it delivered via email.

* **Data Import and Export**: TCExam uses Open Standard protocols for data
filing and interchange: TSV, XML and PDF. Everything is fully documented to
be easily extended or used by external applications. Custom filters can be
added to import data from other systems. Include Optical Mark Recognition
(OMR) system to import users' answers from paper sheets. TCExam supports
several types of remote authentication and single-sign-on protocols: LDAP,
RADIUS, CAS.

* **Multimedia Content**: TCExam uses a common mark-up language to add text
formatting, images, multimedia objects (audio and video) and mathematical
formulas (supports LaTeX and MathML).

* **Unique Tests**: TCExam can simultaneously generate unique tests for
different users by randomly selecting and sorting questions and alternative
answers. This feature drastically reduces or eliminates the chances of
cheating on the test.

* **Paper Testing with Optical Character Recognition (OMR)**: TCExam can
generate printable PDF documents for pen-and-paper testing. The OMR answer
sheet can be scanned and uploaded to TCExam for automatic test importing,
scoring and reporting.

* **Security**: TCExam is more secure than traditional Pen-and-Paper testing.
It supports SSL (Secure Socket Layer) encryption and includes various
authentication levels to discriminate the access to the various sections
(user/password, access level, group, IP address). Test time, test activity,
results, and statistics: every part of the test is managed and stored in the
server, leaving no sensitive data in the users’ client computers.


CHANGELOG.TXT
14.3.1 (2021-05-30)
	- Various security fixes.
	- Perubahan pada seri patch lama.

14.3.0 (2020-06-12)
	- Various security fixes.

14.0.0 (2017-04-22)
	- Change password hashing algorithm.
	- NOTE: all passwords have to be reset!

13.3.0 (2016-09-06)
	- Fix undefined K_USRREG_PASSWORD_RE issue.

13.2.1 (2016-08-22)
	- Fix fatal error in tce_db_dal.php.

13.2.0 (2016-08-21)
	- The K_USRREG_PASSWORD_RE constant was added in shared/config/tce_user_registration.php to set the regular expression used to filter input passwords.
	- An error with the C_mailer class was fixed.
	- Automatic code formatting.

13.1.1 (2016-03-15)
	- Fix a critical bug introcuced in version 13.0.0 that prevent storing the correct answers.

13.0.1 (2015-12-12)
	- Fix a bug related to PDF email attachments.

13.0.0 (2015-12-12)
	- Hide internal answer IDs during test execution.

12.2.6 (2015-10-26)
	- README in markdown format.
	- Updated license terms.

12.2.5 (2015-06-18)
	- TCPDF Library was updated.
	- German translation was updated.

12.2.4 (2015-04-29)
	- TCPDF Library was updated.

12.2.3 (2015-04-28)
	- TCPDF Library was updated.

12.2.2 (2015-03-29)
	- LDAP Authentication bug was fixed.

12.2.1 (2015-03-27)
	- Do not check for brute-forcing login attempts if K_BRUTE_FORCE_DELAY_RATIO is set to zero.
	- The LDAP config now contains 2 new settings for the root DN and password used during binding.
	- A LDAP authentication bug was fixed.

12.1.30 (2015-01-28)
	- TCPDF library was updated.

12.1.29 (2014-12-10)
	- TCPDF library was updated.

12.1.28 (2014-12-09)
	- TCPDF library was updated.

12.1.027 (2014-12-07)
	- TCPDF library was updated.

12.1.026 (2014-11-15)
	- TCPDF library was updated.

12.1.025 (2014-11-08)
	- TCPDF library and fonts were updated.

12.1.024 (2014-06-11)
	- An unused paramer was removed.
	- OMR scan was improved.
	- TCPDF library was updated.

12.1.023 (2014-05-14)
	- Some language typos were fixed.

12.1.022 (2014-05-12)
	- TCPDF library and fonts were updated.

12.1.021 (2014-04-27)
	- Deprecated curly braces substring syntax was replaced with square braces.
	- TCPDF library was updated.

12.1.020 (2014-04-15)
	- TCPDF library was updated.

12.1.019 (2014-03-24)
	- Emulating magic_quotes_gpc for PHP 5.5 compatibility.

12.1.018 (2014-03-04)
	- New Farsi (Persian) language translation.
	- Some LTR direction settings were fixed.

12.1.017 (2014-03-02)
	- The unhtmlentities function was changed for performances and PHP 5.5 compatibility.
	- TCPDF library was updated.

12.1.016 (2014-01-27)
	- Statistics options are now enforced also in various export formats.

12.1.015 (2014-01-26)
	- A bug on session function was fixed.
	- A bug on mysql error reporting was fixed.

12.1.014 (2014-01-23)
	- Test user time was fixed in public reports.

12.1.013 (2014-01-21)
	- Test Results Summary was improved: some statistics are now disabled by default to improve performances.
	- TSV export for results was fixed.
	- File media selector bug was fixed.
	- TCPDF library was updated.

12.1.012 (2013-12-24)
	- MathML is now supported via the [mathml] tag. Note that is not yet possible to export the formulas in PDF format.

12.1.011 (2013-12-11)
	- A problem related to custom error messages and TCPDF library was fixed.

12.1.010 (2013-12-01)
	- TCPDF library was updated.

12.1.009 (2013-10-24)
	- Bug fix related to the new F_escape_sql function signature.

12.1.008 (2013-10-23)
	- The signature of the F_escape_sql function was changed to add the database link identifier.
	- The MySQL Database Abstraction Layer now uses the MySQLi extension by default (you can set the 'K_DATABASE_TYPE' constant to 'MYSQLDEPRECATED' to use the old extension). 
	- TCPDF Library was updated.

12.1.007 (2013-09-14)
	- TCPDF Library was updated.

12.1.006 (2013-09-05)
	- New JSON export format.
	- TSV report now includes the details for each user.

12.1.005 (2013-08-23)
	- Support for 0-difficulty questions was added.

12.1.004 (2013-08-07)
	- A problem with email reports was fixed.

12.1.003 (2013-08-05)
	- TCPDF Library was updated.

12.1.002 (2013-07-16)
	- Missing SQL files were added - please ugrade your existing DB.

12.1.001 (2013-07-14)
	- A problem related to SVG graphs on PDF export was fixed.

12.1.000 (2013-07-10)
	- IMPORTANT: the database schema is changed please use the upgrading SQL files in case of upgrade.
	- This version includes a new feature that allows to restrict test access based on client SSL certificate.
	- It is now possible to restrict the admin and/or public area access only to clients with valid SSL certificate (check the new options on the admin and public configuration files).
	- TCPDF Library was updated.

------------------------------------------------------------


12.0.014 (2013-05-31)
	- TCPDF library was updated.
	- An additional ordering option for questions was added on the test editor.

12.0.013 (2013-05-21)
	- TCPDF library was updated.
	- An error on a SQL query when importing OMR answers was fixed.
	- An option was added to overwrite non-repeatable test answers when importing via OMR.

12.0.012 (2013-05-02)
	- The constant K_USER_GROUP_RSYNC was added on the configuration file /shared/config/tce_config.php to enable or disable the the users groups synchronization when an alternate authentication method is used.

12.0.011 (2013-04-25)
	- OMR import was improved : MCSA with multiple answers are now set as unanswered.

12.0.010 (2013-04-22)
	- Latex rendering class was removed and replaced with a new function.
	- phpCAS class was updated.

12.0.009 (2013-04-15)
	- TCExam project was moved to GitHub: https://github.com/tecnickcom/tcexam/

12.0.008 (2013-04-12)
	- A bug on test selection form was fixed.
	- Default permissions for file manager were lowered to the operator level.
	- Non-administrators may now access to their cache folder or the ones of the users in their groups.
	- Some bugs on TSV question importer were fixed.

12.0.007 (2013-04-09)

admin/code/tce_page_info.php
<?php
//============================================================+
// File name   : tce_page_info.php
// Begin       : 2004-05-21
// Last Update : 2020-05-06
//
// Description : Outputs TCExam information page.
//
// Author: Nicola Asuni
//
// (c) Copyright:
//               Nicola Asuni
//               Tecnick.com LTD
//               www.tecnick.com
//               info@tecnick.com
//
// License:
//    Copyright (C) 2004-2020 Nicola Asuni - Tecnick.com LTD
//    See LICENSE.TXT file for more information.
//============================================================+

/**
 * @file
 * Outputs TCExam information page.
 * @package com.tecnick.tcexam.admin
 * @author Nicola Asuni
 * @since 2004-05-21
 */

/**
 */

require_once('../config/tce_config.php');

$pagelevel = K_AUTH_ADMIN_INFO;
require_once('../../shared/code/tce_authorization.php');

$thispage_title = $l['t_page_info'];
$thispage_title_icon = '<i class="fas fa-info-circle"></i> ';
require_once('../code/tce_page_header.php');

require_once('tce_page_header.php');

echo '<div class="container mb-15 ta-left">'.K_NEWLINE;
echo '<div class="d-block p-15">'.K_NEWLINE;
echo '<span id="tcexam-desc">TCExam Mobile Friendly version is a fork of TCExam by <a href="https://evankumanireng2.blogspot.com">@evan</a>. This version fix several bug in original TCExam and add UI enhancement for public and admin area.</span><br/><hr/>'.K_NEWLINE;
echo '<span id="tcexam-desc">'.$l['d_tcexam_desc'].'</span>'.K_NEWLINE;

echo '<ul class="credits">'.K_NEWLINE;
echo '<li><strong>'.$l['w_author'].':</strong> Nicola Asuni</li>'.K_NEWLINE;
echo '<li><strong>Copyright:</strong><br /> (c) 2004-2020 Nicola Asuni - Tecnick.com LTD<br />'.K_NEWLINE;
echo '<a href="mailto:info@tecnick.com">info@tecnick.com</a> - '.K_NEWLINE;
echo '<a href="http://www.tecnick.com" title="'.$l['m_new_window_link'].'">www.tecnick.com</a></li>'.K_NEWLINE;
echo '<li><strong>'.$l['w_license'].':</strong> <a href="../../LICENSE.TXT" title="'.$l['m_new_window_link'].'">LICENSE.TXT</a></li>'.K_NEWLINE;
echo '</ul>'.K_NEWLINE;

echo '<h2>'.$l['t_third_parties'].'</h2>'.K_NEWLINE;

echo '<p>TCExam includes some third-party software components that are not strictly required but have been included as you convenience, and if used are subject to their respective licenses.</p>'.K_NEWLINE;

echo '<ul class="credits">'.K_NEWLINE;

echo '<li><strong>PHPMailer</strong><br />
Full Featured Email Transfer Class for PHP.<br />
Authors:<ul>
<li>Brent R. Matzelle (original founder)</li>
<li>2014 Marcus Bointon
<li>2010 - 2012 Jim Jagielski
<li>2004 - 2009 Andy Prevost
</ul>
Homepage: <a href="https://github.com/PHPMailer/PHPMailer/">https://github.com/PHPMailer/PHPMailer</a><br />
License: <a href="http://www.gnu.org/copyleft/lesser.html" title="GNU Lesser General Public License">LGPL (GNU LESSER GENERAL PUBLIC LICENSE)</a><br />
Location: /shared/phpmailer/<br /><br />
</li>'.K_NEWLINE;

echo '<li><strong>The DHTML Calendar</strong><br />
Calendar widget written in Javascript.<br />
Author: Mihai Bazon (<a href="mailto:mihai_bazon@yahoo.com">mihai_bazon@yahoo.com</a>)<br />
Homepage: <a href="http://dynarch.com/mishoo/" title="mishoo">http://dynarch.com/mishoo/</a><br />
License: <a href="http://www.gnu.org/copyleft/lesser.html" title="GNU Lesser General Public License">LGPL (GNU LESSER GENERAL PUBLIC LICENSE)</a><br />
Location: /shared/jscripts/jscalendar/<br /><br />
</li>'.K_NEWLINE;

echo '<li><strong>Javascript VirtualKeyboard</strong><br />
Author: Ilya Lebedev (<a href="mailto:ilya@lebedev.net">ilya@lebedev.net</a>)<br />
Homepage: <a href="http://debugger.ru/projects/virtualkeyboard" title="VirtualKeyboard">http://debugger.ru/projects/virtualkeyboard</a><br />
License: <a href="http://www.gnu.org/copyleft/lesser.html" title="GNU Lesser General Public License">LGPL (GNU LESSER GENERAL PUBLIC LICENSE)</a><br />
Location: /shared/jscripts/vk/<br /><br />
</li>'.K_NEWLINE;

echo '<li><strong>Radius Class</strong><br />
Radius client implementation in pure PHP.<br />
Author: SysCo/al (<a href="mailto:developer@sysco.ch">developer@sysco.ch</a>)<br />
Homepage: <a href="http://developer.sysco.ch/php/" title="sysco">http://developer.sysco.ch/php/</a><br />
License: <a href="http://www.gnu.org/copyleft/lesser.html" title="GNU Lesser General Public License">LGPL (GNU LESSER GENERAL PUBLIC LICENSE)</a><br />
Location: /shared/radius/<br /><br />
</li>'.K_NEWLINE;

echo '<li><strong>phpCAS</strong><br />
CAS client in PHP.<br />
Copyright: 2007-2015, JA-SIG, Inc. <a href="http://www.jasig.org" title="Jasig Website">http://www.jasig.org</a><br />
Homepage: <a href="https://wiki.jasig.org/display/CASC/phpCAS" title="phpCAS">https://wiki.jasig.org/display/CASC/phpCAS</a><br />
License: <a href="http://www.apache.org/licenses/LICENSE-2.0" title="Apache License, Version 2.0">Apache License, Version 2.0</a><br />
Location: /shared/cas/<br /><br />
</li>'.K_NEWLINE;

echo '<li><strong>Fonts</strong><br />

TCExam includes fonts for the <a href="http://www.tcpdf.org" title="TCPDF - PHP class for PDF">TCPDF</a> library, they are not strictly required but have been included as you convenience.<br />
<br />
All the PHP files on the fonts directory are subject to the general TCPDF license (GNU-LGPLv3), they do not contain any binary data but just a description of the general properties of a particular font. These files can be also generated on the fly using the font utilities and TCPDF methods.
<br />
All the original binary TTF font files have been renamed for compatibility with TCPDF and compressed using the gzcompress PHP function that uses the ZLIB data format (.z files).<br />
<br />
The binary files (.z) that begins with the prefix "free" have been extracted from the GNU FreeFont collection (GNU-GPLv3).<br />
The binary files (.z) that begins with the prefix "pdfa" have been derived from the GNU FreeFont, so they are subject to the same license.<br />
For the details of Copyright, License and other information, please check the files inside the directory fonts/freefont-20100919<br />
Homepage: <a href="http://www.gnu.org/software/freefont/" title="GNU FreeFont">http://www.gnu.org/software/freefont/</a><br />
License: <a href="http://www.gnu.org/licenses/gpl.html" title="GNU General Public License, version 3">GNU-GPLv3</a><br />
<br />
The binary files (.z) that begins with the prefix "dejavu" have been extracted from the DejaVu fonts 2.33 (Bitstream) collection.<br />
For the details of Copyright, License and other information, please check the files inside the directory fonts/dejavu-fonts-ttf-2.33<br />
Homepage: <a href="http://dejavu-fonts.org" title="DejaVu fonts">http://dejavu-fonts.org</a><br />
License: <a href="http://dejavu-fonts.org/wiki/License" title="Bitstream License">DejaVu changes are in public domain</a><br />
<br />
The binary files (.z) that begins with the prefix "ae" have been extracted from the Arabeyes.org collection (GNU-GPLv2).<br />
Homepage: <a href="http://projects.arabeyes.org" title="Arabeyes.org">http://projects.arabeyes.org</a><br />
License: <a href="http://www.gnu.org/licenses/old-licenses/gpl-2.0.html" title="GNU General Public License, version 2">GNU-GPLv2</a><br />


</li>'.K_NEWLINE;

echo '</ul>'.K_NEWLINE;

echo '<h2>'.$l['t_translations'].'</h2>'.K_NEWLINE;

echo '<ul class="credits">'.K_NEWLINE;
echo '<li>[AR] Arabic : Red Sea</li>'.K_NEWLINE;
echo '<li>[AZ] Azerbaijani : Jamil Farzana</li>'.K_NEWLINE;
echo '<li>[BG] Bulgarian : Georgi Kostadinov</li>'.K_NEWLINE;
echo '<li>[BR] Brazilian Portuguese : Carlos Eduardo Vianna, Flávio Veras</li>'.K_NEWLINE;
echo '<li>[CN] Chinese : Liu Yongxin, Zheng Xiaojing</li>'.K_NEWLINE;
echo '<li>[DE] German : Oliver Kasch, André Scherrer, Wolfgang Stöggl</li>'.K_NEWLINE;
echo '<li>[EL] Greek : Kottas Alexandros</li>'.K_NEWLINE;
echo '<li>[EN] English : Nicola Asuni</li>'.K_NEWLINE;
echo '<li>[ES] Spanish : Carlos Alarcon, Maria del Rocio Peñas Serrano, Alejandra Ruiz</li>'.K_NEWLINE;
echo '<li>[FA] Farsi (Persian): Mahmoud Saghaei</li>'.K_NEWLINE;
echo '<li>[FR] French : Roger Koukerjinian, André Scherrer</li>'.K_NEWLINE;
echo '<li>[HI] Hindi : Mahesh K Bhandari, Shekhar K Maravi, Pradeep K Nayak</li>'.K_NEWLINE;
echo '<li>[HE] Hebrew : Oron Peled</li>'.K_NEWLINE;
echo '<li>[HU] Hungarian : Peter Ivanyi, Tibor Balázs</li>'.K_NEWLINE;
echo '<li>[ID] Indonesian : Mr Ahmad </li>'.K_NEWLINE;
echo '<li>[IT] Italian : Nicola Asuni</li>'.K_NEWLINE;
echo '<li>[JP] Japanese : Koji Nakajima</li>'.K_NEWLINE;
echo '<li>[MR] Marathi : Tushar Sayankar</li>'.K_NEWLINE;
echo '<li>[MS] Malay (Bahasa Melayu) : Arvind Prakash Jha</li>'.K_NEWLINE;
echo '<li>[NL] Dutch : Chris de Boer</li>'.K_NEWLINE;
echo '<li>[PL] Polish : Tomasz Parol</li>'.K_NEWLINE;
echo '<li>[RO] Romanian : Ovidiu Dragomir</li>'.K_NEWLINE;
echo '<li>[RU] Russian : Andrey, Sergey C.</li>'.K_NEWLINE;
echo '<li>[TR] Turkish : Mehmet Arif Icir</li>'.K_NEWLINE;
echo '<li>[UR] Urdu : Ghulam Abbas</li>'.K_NEWLINE;
echo '<li>[VN] Vietnamese : Nguyen Quynh Nga</li>'.K_NEWLINE;
echo '</ul>'.K_NEWLINE;

echo '</div>'.K_NEWLINE;

echo '<br />'.K_NEWLINE;

// display credit logos
echo '<div class="creditslogos">'.K_NEWLINE;
echo '<a href="http://www.tcexam.org/license.php"><img src="../../images/credits/agplv3-88x31.png" alt="TCExam License" width="88" height="31" style="border:none;" /></a>'.K_NEWLINE;
//echo '<a href="http://www.php.net"><img src="../../images/credits/poweredby_php_88x31.png" alt="Powered by PHP" width="88" height="31" /></a>'.K_NEWLINE;
//echo '<a href="http://www.mysql.com"><img src="../../images/credits/poweredbymysql_88x31.png" alt="Powered by MySQL" width="88" height="31" /></a>'.K_NEWLINE;
//echo '<a href="http://www.postgresql.org"><img src="../../images/credits/poweredbypgsql_88x31.png" alt="Powered by PostgreSQL" width="88" height="31" /></a>'.K_NEWLINE;
echo '<a href="http://validator.w3.org/check?uri='.K_PATH_HOST.$_SERVER['SCRIPT_NAME'].'" title="This Page Is Valid XHTML 1.0 Strict!"><img src="../../images/credits/w3c_xhtml10_88x31.png" alt="Valid XHTML 1.0!" height="31" width="88" style="border:none;" /></a>'.K_NEWLINE;
echo '<a href="http://jigsaw.w3.org/css-validator/" title="This document validates as CSS!"><img src="../../images/credits/w3c_css_88x31.png" alt="Valid CSS1!" height="31" width="88" style="border:none;" /></a>'.K_NEWLINE;
echo '<a href="http://www.w3.org/WAI/WCAG1AAA-Conformance" title="Explanation of Level Triple-A Conformance"><img src="../../images/credits/w3c_wai_aaa_88x31.png" alt="Level Triple-A conformance icon, W3C-WAI Web Content Accessibility Guidelines 1.0" width="88" height="31" style="border:none;" /></a>'.K_NEWLINE;
echo '</div>'.K_NEWLINE;
echo '</div>'.K_NEWLINE;

require_once('tce_page_footer.php');

//============================================================+
// END OF FILE
//============================================================+
